AIAgent/common/config/validation_config.py

```diff
@@ -0,0 +1,36 @@
+from abc import ABC
+from typing import Literal, Union
+
+from common.config.platform_config import Platform
+from pydantic import Field
+from pydantic.dataclasses import dataclass as pydantic_dataclass
+
+
+@pydantic_dataclass
+class Validation(ABC):
+    val_type: str
+
+
+@pydantic_dataclass
+class ValidationLoss(Validation):
+    val_type: Literal["loss"]
+    batch_size: int
+
+
+@pydantic_dataclass
+class ValidationSVM(Validation, ABC):
+    platforms_config: list[Platform] = Field(alias="PlatformsConfig")
+    fail_immediately: bool = Field(default=False)
+
+
+@pydantic_dataclass
+class ValidationSVMViaServer(ValidationSVM):
+    val_type: Literal["svms_server"]
```

Comment:
should be set in `__post_init__`

Reply:
I'm not sure I understand. The documentation says that the discriminator should be specified as a literal attribute. https://docs.pydantic.dev/latest/concepts/unions/#discriminated-unions-with-callable-discriminator

Reply:
![image](https://github.com/user-attachments/assets/7fdababb-7195-44c1-a1ab-9e8b50c0eebf)

Reply:
ok